corehq/apps/reports/generic.py

```diff
@@ -660,9 +659,13 @@ def export_response(self):
             export_all_rows_task.delay(self.__class__, self.__getstate__())
             return HttpResponse()
         else:
-            temp = io.BytesIO()
-            export_from_tables(self.export_table, temp, self.export_format)
-            return export_response(temp, self.export_format, self.export_name)
+            self._export_response_direct()
```

Comment:
nit: might consider adding a comment explaining why it is like this

Reply:
Done 👍

Reply:
This must return a value. Otherwise Django will raise `ValueError: The view ... didn't return an HttpResponse object. It returned None instead.` Maybe this should have returned?
```py
return self._export_response_direct()
```
In any case, I'm going to roll this PR back. Would have been good to add an automated test to ensure `export_response` always returns a value.

Reply:
This was an oversight from my side @millerdev, apologies for that. I tested for only the reports that were coming to email and not for the others. 

I would add tests for it in the next PR.

Reply:
sorry I also should have caught that in review! :disappointed: